src/compiler/checker.ts

```diff
@@ -13274,13 +13274,19 @@ namespace ts {
                         undefined;
         }
 
+        function isUncalledFunctionReference(node: Node, symbol: Symbol) {
+            return !(symbol.flags & SymbolFlags.Function)
+                || !isCallExpression(findAncestor(node, n => !isAccessExpression(n)) ?? node.parent)

```

Comment:
NewExpression?

Reply:
Or an invocationexpression in general

Just switch to `||`, it's not a primitive type